### DISCUSSION TITLE
How to read MIDI events in real time ?

### DISCUSSION THREAD

**Discussion by essemo** [GrandOrgue/grandorgue#2067] [Help]
Dear all, hello.
I express again my great gratitude to the developers for the exceptional work you do with GrandOrgue.

I share with you a difficulty that I encounter with Ubuntu Studio 24.04, (that I do not encounter with version 22.04) to read MIDI events in real time.

Under Ubuntu Studio 22.04 I like to use Kmidimon: once Kmidimon is launched, it appears in the MIDI OUTPUT DEVICES of GrandOrgue.

But under 24.04, nothing appears in the MIDI OUTPUT DEVICES;
and another curiosity: the Graph or Patchance does not display the MIDI OUTPUT of Grandorgue (while it does appear in the Graph, under 22.04).
I activated the LOG MIDI EVENTS of the AUDIO/MIDI menu but nothing appears on the computer screen...

Thank you very much for the help you can give me.
Please accept my respectful and cordial greetings.
Essemo

NB: Under 22.04 : I use grandorgue_3.15.3-1_amd64.deb
and under 24.04 : grandorgue-wx32_3.14.2-1_amd64.deb

**Comment by oleg68** [comment]
@essemo I suggest you to install qjackctl, Do you see anything in the Graph?

**Comment by essemo** [comment]
Hello,Thank you for your interest in my problem...
Under Ubuntu Studio 24.04 QjackCtl is installed correctly.
My surprise is precisely to note that on the Graph, the audio outputs of GrandOrgue appear, but not the MIDI outputs or inputs... while they appear correctly in the Graph of QjackCtl under Ubuntu Studio 22.04...

    Le dimanche 8 décembre 2024 à 14:45:44 UTC+1, Oleg Samarin ***@***.***> a écrit :  
 
 


@essemo I suggest you to install qjackctl, Do you see anything in the Graph?

**Comment by essemo** [comment]
A small correction and a precision:
MIDI inputs appear, but it is the outputs that are invisible.
Here is the Graph when I launch Kmidimon...
Another problem: when I check "log MIDI events" in the AUDIO/MIDI menu, I get nothing... my MIDI events such as the ControlChange sends that I have set up, are not displayed...
    Le dimanche 8 décembre 2024 à 14:45:44 UTC+1, Oleg Samarin ***@***.***> a écrit :  
 
 


@essemo I suggest you to install qjackctl, Do you see anything in the Graph?

**Comment by essemo** [comment]
I apologize: the screenshot was bad...

    Le dimanche 8 décembre 2024 à 14:45:44 UTC+1, Oleg Samarin ***@***.***> a écrit :  
 
 


@essemo I suggest you to install qjackctl, Do you see anything in the Graph?

**Comment by essemo** [comment]
Dear Sir, hello.
I just did a test with AppImage ( grandorgue-3.15.3-1.x86_64.AppImage ) version: everything works fine! The MIDI devices are all recognized!

The problem seems to come from the version I installed:
grandorgue-wx32_3.15.3-1_amd64.deb

I did an autoremove then reinstalled the application, but the problem persists: the MIDI devices are invisible in the Settings / MIDI Devices window...

Maybe you have a solution?
In any case, thank you for your help.
Best regards
Jean-Bernard

    Le dimanche 8 décembre 2024 à 14:45:44 UTC+1, Oleg Samarin ***@***.***> a écrit :  
 
 


@essemo I suggest you to install qjackctl, Do you see anything in the Graph?

**Reply by oleg68** [reply]
> I did an autoremove then reinstalled the application, but the problem persists: the MIDI devices are invisible in the Settings / MIDI Devices window...

Could you post the screenshots from File->Settings->Midi from grandorgue-wx32_3.15.3-1_amd64.deb and grandorgue-3.15.3-1.x86_64.AppImage here?

**Comment by essemo** [comment]
Hello dear Sir,
here are the screenshots :

    Le mardi 10 décembre 2024 à 20:49:30 UTC+1, Oleg Samarin ***@***.***> a écrit :  
 
 



I did an autoremove then reinstalled the application, but the problem persists: the MIDI devices are invisible in the Settings / MIDI Devices window...


Could you post the screenshots from File->Settings->Midi from grandorgue-wx32_3.15.3-1_amd64.deb and grandorgue-3.15.3-1.x86_64.AppImage here?

**Comment by oleg68** [comment]
I cannot see your screenshot. Have you really posted it?

ср, 11 дек. 2024 г., 11:44 essemo ***@***.***>:

**Reply by essemo** [reply]
Here are the screenshots (sorry, I made a mistake...)[
![Settings for grandorgue-wx32_3 15 3-1_amd64](https://github.com/user-attachments/assets/0584db1d-b432-467d-b0cb-9a17dca62dc0)
![Settings for grandorgue-3 15 3-1 x86_64 AppImage](https://github.com/user-attachments/assets/e19f793d-b314-4513-bc92-9dfe1e502a0c)
